### DISCUSSION TITLE
Structured Output Converter

### DISCUSSION THREAD

**Discussion by lordofthejars** [langchain4j/langchain4j#2025] [Ideas]
Hi, today I have read that Spring AI implements Structured Output Converter (https://docs.spring.io/spring-ai/reference/api/structured-output-converter.html) and I think it is really good idea to adopt. 

I know that for JSON it is supported with https://docs.langchain4j.dev/integrations/language-models/google-ai-gemini/#response-format--response-schema 

Implementing this feature upstream will allow developers to use a similar approach for any other formats we provide (not only JSON) and will also allow them to implement it on their own.

**Comment by smileyboy2019** [comment]
Suggest SPI functionality, as many companies have different output requirements
